rdrf/rdrf/admin.py

```diff
@@ -178,6 +183,93 @@ def get_readonly_fields(self, request, obj=None):
         return () if obj is None else ("code",)
 
 
+class ActivationKeyExpirationListFilter(admin.SimpleListFilter):
+    title = _('activation key expired')
+    parameter_name = 'activation_key_expired'
+
+    def lookups(self, request, model_admin):
+        return [
+            ('True', _('True')),
+            ('False', _('False')),
+        ]
+
+    def queryset(self, request, queryset):
+        expired_emails = []
+        unexpired_emails = []
+        for profile in queryset:
+            user_email = profile.user.email
+            if profile.activation_key_expired():
+                expired_emails.append(user_email)
+            else:
+                unexpired_emails.append(user_email)
+
+        if self.value() == 'True':
+            return RegistrationProfile.objects.filter(user__email__in=expired_emails)
+        if self.value() == 'False':
+            return RegistrationProfile.objects.filter(user__email__in=unexpired_emails)
+
+
+def resend_activation_mail(email, site, request=None):
+    user = RegistrationProfile.objects.filter(user__email__iexact=email)
+    if not user.exists():
+        return False
+
+    user_profile = user.first()
+    if user_profile.activated:
+        return False
+
+    user_profile.create_new_activation_key()
+    user_profile.send_activation_email(site, request)
+
+    return True
+
+
+def activate_user(activation_key):
+    sha256_re = re.compile('^[a-f0-9]{40,64}$')
+
+    def activate(user_profile):
```

Comment:
I'm concerned about these functions being `public`, can we nest them inside the class they are used or at least rename with an underscore e.g. `_activate` so it is clear the intention is not to use them outside of this admin script?